### DISCUSSION TITLE
Не все сегменты матрицы работают

### DISCUSSION THREAD

**Discussion by Ilich-ism** [vvip-68/LedPanelWiFi#32] [Вопросы и ответы]
Всем привет.
Собрал матрицу на d1 mini, питание 8А. Матрица состоит из 3-х сегментов 9x11. Т.е. всего 297 диодов.
3 гирлянды просто соединены последовательно.
Если все 3 сегмента запустить, то на матрице много белых или другого цвета диодов, которые горят статично. Даже если  оставить подключенными только 2 гирлянды, но в настройках поменять на 3 сегмента.
Если настроить на 2 сегмента, то все работает как надо.
В чем может быть причина?
Спасибо

**Comment by vvip-68** [comment]
А если они соединены последовательно и однотипно - если поставить 27x11 и обозначить это как один сегмент - что-то меняется?

**Reply by hostaway-iosipov** [reply]
Спасибо за совет. Так вроде работает, теперь буду разбираться как это замапить, чтобы отображалось все правильно)

**Reply by vvip-68** [reply]
Ну, если возникает необходимость мапить - значит один сегмент не является продолжением другого (вы писали " 3 гирлянды соединены последовательно")...
Тут или как у вас бить на сегменты или использовать карту индексов - отдельный файл с массивом номеров светодиодов, определяющий их физический порядок на матрице.

Про маппинг (карту индексов)  - тут - https://github.com/vvip-68/LedPanelWiFi/wiki/Сборная-матрица,-состоящая-из-матриц-разного-типа-и-размера#сборная-матрица-из-трех-сегментов-16х16-и-32x8

Имейте в виду, если есть SD-карта и эффекты из Jinx - ролики грабятся с учетом карты индексов (карты физического расположения), но утилита обрезки роликов пока не поддерживает работу с картами индексов - обрезать такие ролики не получится...

Хотя я не понимаю как количество сегментов влияет на свечение диодов белым цветом в беспорядке.

**Reply by hostaway-iosipov** [reply]
У меня соединены 3 гирлянды и получается так, что на одном участке перевернуто изображение, т.к. гирлянды так соединены и расположены на окне. Это возможно исправить маппингом?
SD карту и jinx использовать не буду, но спасибо за подсказку.

**Reply by vvip-68** [reply]
Да, карта индексов поможет в данном случае

**Reply by Ilich-ism** [reply]
Все получилось. Скорее всего эти рандомные участки были связаны с плохими контактами при соединении гирлянд.
Появился еще вопрос, со временем (минут 15-20) FPS сильно проседает и все эффекты тормозят. Это особенности wemos d1 mini или что-то с этим можно сделать?

**Reply by vvip-68** [reply]
Ничего со временем проседать не должно. Ни на Wemos ни на ESP32.
Несколько людей уже писали про это. Или один и тот же человек в разных форумах (?)
У меня такого не наблюдается. А поскольку я эту "ошибку" повторить не могу - я и исправить ее не могу, даже если она реально существует. Вон гирлянды на окне уже почти месяц не выключаясь работают. Всё бегает, ничего не тормозит...
